src/Controls/src/Core/Platform/GestureManager/GestureManager.iOS.cs

```diff
@@ -563,5 +599,51 @@ void OnElementChanged(object sender, VisualElementChangedEventArgs e)
 				}
 			}
 		}
+
+		ButtonsMask AddFakeRightClickForMacCatalyst(IGestureRecognizer recognizer)
```

Comment:
Most the changes in this file are organizational. There seemed to be a lot of duplicated code added to process spans. I needed that same code generalized so I could use it from a `UIContextMenu`. 

The code here is the main code required to make right click fire an event on catalyst.